### DISCUSSION TITLE
invalid_auth

### DISCUSSION THREAD

**Discussion by mcepl** [wee-slack/wee-slack#928] [Q&A] [answered]
When running a [slightly modified](https://git.cepl.eu/cgit/toys/wee-slack/commit/?h=build&id=4ec0d658b5051bdcffc7a20530003d323967ff90) script, I get this in `#weechat` channel on start after storing session token:

```
12:56 ERROR: Failed connecting to Slack with token xoxc-2278127410...Qugg%3D%3D: channels: invalid_auth, members: invalid_auth, usergroups: invalid_auth, prefs: invalid_auth, presence: invalid_auth, channels: invalid_auth
12:56 ERROR: Failed connecting to Slack with token xoxc-2278127410...OQUDBo%3D;: channels: invalid_auth, channels: invalid_auth, usergroups: invalid_auth, prefs: invalid_auth, presence: invalid_auth, members: invalid_auth
```

Any ideas?

**Comment by trygveaa** [comment]
Judging from the error message it looks like it includes the cookie in the token instead of separating them properly. Are you sure you have a `:` between the token and the cookie?

**Reply by mcepl** [reply]
```
tumbleweed-pkg~/.c/weechat (master *)$ grep token plugins.conf
python.slack.slack_api_token = "xoxc-2278...3:xoxd-mPhZ...3D;"
python.slack.slack_api_token = "List of Slack API tokens, one per Slack instance you want to connect to; see the README for details on how to get these (note: content is evaluated, see /help eval)."
tumbleweed-pkg~/.c/weechat (master *)$ 
```
(of course, both tokens are abbreviated)

**Reply by itorres** [reply, marked as answer]
Hi @mcepl,

What has worked for me is (from the [readme](https://raw.githubusercontent.com/wee-slack/wee-slack/master/README.md)):

> Note that in some cases it may be necessary to include the `d-s` cookie as
well. If so, you can supply it in this format `<token>:d=<d_cookie>;d-s=<d-s_cookie>`.

I had to do a good bunch of trial and errors so this litany was useful to reset the token:
```/set plugins.var.python.slack.slack_api_token ""
/python reload slack
/slack register xoxc-<token>:d=xoxd-<token>;d-s=1234567890
/python reload slack
```

Then I got in "timeout error" territory which was fixed by following the instructions in [this comment](https://github.com/wee-slack/wee-slack/issues/877#issuecomment-1810932287).
The tl;dr is to checkout the branch `only-load-joined-conversations` rebasing it on master.
1. `$ git clone git@github.com:wee-slack/wee-slack.git; cd wee-slack`
2. `$ git checkout only-load-joined-conversations`
3. `$ git rebase master`
4. `$ ln wee_slack.py ~/.weechat/pythonslack.py`
5. `$ ln -s ~/.weechat/python/slack.py ~/.weechat/python/autoload/slack.py`

@trygveaa out of curiosity wouldn't the behaviour on that branch be the preferred one? Why is it not merged on master, does it break anything known?

Thanks a lot to the maintainers and the community for the steps.

**Reply by mcepl** [reply]
Thank you very much!
